pkg/render/prometheus_service.go

```diff
@@ -0,0 +1,253 @@
+	}
+}
+
+type tigeraPrometheusServiceComponent struct {
+	installation           *operator.InstallationSpec
+	pullSecrets            []*corev1.Secret
+	prometheusServicePort  int
+	prometheusServiceImage string
+}
+
+func (p *tigeraPrometheusServiceComponent) ResolveImages(is *operator.ImageSet) error {
+	reg := p.installation.Registry
+	path := p.installation.ImagePath
+	prefix := p.installation.ImagePrefix
+
+	prometheusServiceImage, err := components.GetReference(components.ComponentTigeraPrometheusService, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+
+	p.prometheusServiceImage = prometheusServiceImage
+
+	return nil
+}
+
+// Objects returns the lists of objects in this component that should be created and/or deleted during
+// rendering.
+func (p *tigeraPrometheusServiceComponent) Objects() (objsToCreate, objsToDelete []client.Object) {
+	// tigera-prometheus-objects
+	namespacedObjects := []client.Object{}
+
+	// place pullsecrets secrets under tigera-prometheus
+	secrets := secret.CopyToNamespace(common.TigeraPrometheusNamespace, p.pullSecrets...)
+	namespacedObjects = append(namespacedObjects, secret.ToRuntimeObjects(secrets...)...)
+
+	namespacedObjects = append(
+		namespacedObjects,
+		p.tigeraPrometheusServiceDeployment(),
+		p.calicoNodePrometheusService(),
+	)
+
+	objsToCreate = []client.Object{}
+	objsToCreate = append(objsToCreate, namespacedObjects...)
+
+	objsToDelete = []client.Object{}
+
+	return objsToCreate, objsToDelete
+
+}
+
+func (p *tigeraPrometheusServiceComponent) Ready() bool {
+	return true
+}
+
+func (p *tigeraPrometheusServiceComponent) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
+}
+
+// calicoNodePrometheusService sets up a service for the Prometheus Service/Proxy deployment
+func (p *tigeraPrometheusServiceComponent) calicoNodePrometheusService() *corev1.Service {
+
+	s := &corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      calicoNodePrometheusServiceName,
+			Namespace: common.TigeraPrometheusNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Selector: map[string]string{
+				"k8s-app": tigeraPrometheusServiceName,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "web",
+					Port:       PrometheusDefaultPort,
+					Protocol:   corev1.ProtocolTCP,
+					TargetPort: intstr.FromInt(p.prometheusServicePort),
+				},
+			},
+		},
+	}
+
+	return s
+}
+
+// tigeraPrometheusServiceDeployment deployment for the Prometheus Service/Proxy pod and image
+func (p *tigeraPrometheusServiceComponent) tigeraPrometheusServiceDeployment() *appsv1.Deployment {
+	var replicas int32 = 1
+	podDnsPolicy := corev1.DNSClusterFirst
+	podHostNetworked := false
+
+	// set hostnetworked if EKS with Calico CNI due to EKS to allow other Pods to
+	// connecto to tigera-prometheus-service since Calico does not manage node network
+	// in EKS - uses Amazon VPC
```

Comment:
nit: some typos in this comment. WDYTA "Set to host networked if EKS with Calico CNI because the EKS managed Kubernetes APIserver cannot reach the pod network in this configuration. This is because Calico cannot manage the managed control plane nodes' network" or something like that.